CorePayments/src/main/java/com/paypal/android/corepayments/API.kt

```diff
@@ -33,15 +35,29 @@ class API internal constructor(
         return http.send(httpRequest)
     }
 
+    /**
+     * Retrieves the merchant's clientID either from the local cache, or via an HTTP request if not cached.
+     * @return Merchant clientID.
+     */
     @Throws(PayPalSDKError::class)
-    suspend fun getClientId(): String {
+    suspend fun fetchCachedOrRemoteClientID(): String {
+        configuration.accessToken?.let { accessToken ->
+            clientIDCache.get(accessToken)?.let { cachedClientID ->
+                return cachedClientID
```

Comment:
This might not return for the outer suspend function, I can't remember how it works in Kotlin though.

Reply:
Potentially yeah. It's trickier with the nested closures. Technically those `let` functions do return a value though even though we're not capturing the output here. Something like
```kotlin
var clientID = configuration.accessToken?.let { clientIDCache.get(it) }
```
would be equivalent.

Reply:
I'm a little confused - so you're saying there's a potential that `configuration.accessToken?.let` is returning the `accessToken` from the `fetchCachedOrRemoteClientID()` func, unless we specify a variable for it?

Does Kotlin not always require explicit `return`s?

Reply:
In some cases Kotlin doesn't require `return`. In this case the compiler won't warn about the statement on line 44 returning a value that isn't used. In Swift we'd be required to put `let _ =` at the beginning of the line to get it to compile.

Reply:
You are right though this may be less of a concern once that PR is merged.

Reply:
Running through the flow/tests, the function doesn't return the `accessToken` value if it is present. Also if the `accessToken` or `clientIDCache` checks return `null`, it does continue on to execute the actual HTTP fetch below.

So I think we are good here.